sections/futures/MarketInfoBox/MarketInfoBox.tsx

```diff
@@ -69,55 +69,56 @@ const MarketInfoBox: React.FC = () => {
 	}, [commitDeposit, marketInfo?.keeperDeposit]);
 
 	const getPotentialAvailableMargin = (
-		previewTrade: FuturesPotentialTradeDetails | null,
+		trade: FuturesPotentialTradeDetails | null,
 		marketMaxLeverage: Wei | undefined
 	) => {
 		let inaccessible;
 
-		inaccessible = previewTrade?.notionalValue.div(marketMaxLeverage).abs() ?? zeroBN;
+		inaccessible =
+			(marketMaxLeverage && trade?.notionalValue.div(marketMaxLeverage).abs()) ?? zeroBN;
 
 		// If the user has a position open, we'll enforce a min initial margin requirement.
 		if (inaccessible.gt(0)) {
-			if (inaccessible.lt(previewTrade?.minInitialMargin ?? zeroBN)) {
-				inaccessible = previewTrade?.minInitialMargin ?? zeroBN;
+			if (inaccessible.lt(trade?.minInitialMargin ?? zeroBN)) {
+				inaccessible = trade?.minInitialMargin ?? zeroBN;
 			}
 		}
 
 		// check if available margin will be less than 0
-		return previewTrade?.margin?.sub(inaccessible).gt(0)
-			? previewTrade?.margin?.sub(inaccessible).abs()
-			: zeroBN;
+		return trade?.margin?.sub(inaccessible).gt(0) ? trade?.margin?.sub(inaccessible).abs() : zeroBN;
 	};
 
 	const previewAvailableMargin = React.useMemo(() => {
 		const potentialAvailableMargin = getPotentialAvailableMargin(
-			previewTrade,
+			potentialTrade.data,
 			marketInfo?.maxLeverage
 		);
 		return isNextPriceOrder
 			? potentialAvailableMargin?.sub(totalDeposit) ?? zeroBN
 			: potentialAvailableMargin;
-	}, [previewTrade, marketInfo?.maxLeverage, isNextPriceOrder, totalDeposit]);
+	}, [potentialTrade.data, marketInfo?.maxLeverage, isNextPriceOrder, totalDeposit]);
 
 	const previewTradeData = React.useMemo(() => {
-		const size = wei(tradeSize || zeroBN);
+		const size = wei(nativeSize || zeroBN);
 
-		const potentialMarginUsage = previewTrade?.margin.gt(0)
-			? previewTrade?.margin?.sub(previewAvailableMargin)?.div(previewTrade?.margin)?.abs() ??
-			  zeroBN
+		const potentialMarginUsage = potentialTrade.data?.margin.gt(0)
+			? potentialTrade.data?.margin
+					?.sub(previewAvailableMargin)
+					?.div(potentialTrade.data?.margin)
+					?.abs() ?? zeroBN
 			: zeroBN;
 
 		const potentialBuyingPower =
 			previewAvailableMargin?.mul(maxLeverage ?? zeroBN)?.abs() ?? zeroBN;
 
 		return {
 			showPreview: size && !size.eq(0),
-			totalMargin: previewTrade?.margin || zeroBN,
+			totalMargin: potentialTrade.data?.margin || zeroBN,
 			availableMargin: previewAvailableMargin.gt(0) ? previewAvailableMargin : zeroBN,
 			buyingPower: potentialBuyingPower.gt(0) ? potentialBuyingPower : zeroBN,
 			marginUsage: potentialMarginUsage.gt(1) ? wei(1) : potentialMarginUsage,
 		};
-	}, [tradeSize, previewTrade?.margin, previewAvailableMargin, maxLeverage]);
+	}, [nativeSize, potentialTrade.data?.margin, previewAvailableMargin, maxLeverage]);
 
 	return (
 		<StyledInfoBox
```

Comment:
A general comment - I would prefer that the preview doesn't show if the trade size is zero. This is a bit distracting:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/10401554/188014553-6588c125-7620-4e75-b076-aa39a90b978e.png">

Also, a "clear" function for the trade size might be nice if I have a trade preview showing and want to simply view my position.

Reply:
setting to zero now clears preview